values.go

```diff
@@ -1453,6 +1453,11 @@ func NewMeteredDictionary(
 func (Dictionary) isValue() {}
 
 func (v Dictionary) Type() Type {
+	if v.DictionaryType == nil {
```

Comment:
Took me a moment to realize why this is needed: Go creates a typed nil without it. (Typed nil strikes again!) Too bad this is necessary, but seems right, given that allows us to keep the `Type() Type` function.